tests/test_functions.py

```diff
@@ -0,0 +1,119 @@
+import requests, difflib, sys
+import re
+
+command_line_arguments = []
+
+number_of_errors = 0
+
+
+def set_command_line_arguments(arguments):
+    global command_line_arguments
+    command_line_arguments = arguments
+    if "-resetalltests" in command_line_arguments:
+        test_print("resetting all tests by saving every request for future comparisons. You should not have run this unless you are working on the test suite implementation itself and have verified that all tests passed before the reset.")
+
+def test_print(string):
+    sys.stdout.write("[synbiohub test] ")
+    sys.stdout.write(string)
+    sys.stdout.write("\n")
+
+# make html a little more human readable
+def format_html(htmlstring):
+    # for two angle brackets, insert a new line
+    newstring = re.sub('><', '>\n<', htmlstring)
+    # for an ending angle bracket and then something else, enter a new line
+    newstring = re.sub('>(.)', r'>\n\1', newstring)
+    # for something else then an angle bracket, also insert a new line
+    newstring = re.sub('(.)<', r'\1\n<', newstring)
+    
+    return newstring
+
+# perform a get request
+def get_request(request):
+    response = requests.get('http://localhost:7777/' + request)
```

Comment:
Not required right now, but tests should allow URL to be specified as a command-line arg

Reply:
Yep- I wrote a todo in the comment at the top of the pull request. It's on my list.

Reply:
Oh, I meant something different. I mean I should be able to run these tests against `https://synbiohub.org/` or `http://localhost:7371` or any other URL.

Reply:
Oh I see- good idea.